module/mosh.js

```diff
@@ -933,4 +933,59 @@ async function noShipSelected() {
     //log what was done
     console.log(`Told the user that no character was selected.`);
   });
+}
+
+
+//get item from world or compendiums by id or UUID.
+export async function fromIdUuid(id_uuid, options={}){
+  let type = options.type;
+  let item = await fromUuid(id_uuid,options);
+  if(item != null){
+    //we found the item with the id, it probably was an uuid.
+    return item;
+  }
+
+  //we need to manualy find the item:
+  let currentLocation = '';
+  let objectLocation = '';
+  //loop through each compendium
+  game.packs.forEach(function(pack){ 
+    //is this a pack of rolltables?
+    if (pack.metadata.type === type) {
+      //log where we are
+      currentLocation = pack.metadata.id;
+      //loop through each pack to find the right table
+      pack.index.forEach(function(table) { 
+        //is this our table?
+        if (table._id === id_uuid) {
+          //grab the table location
+          objectLocation = currentLocation;
+        }
+      });
+    }
+  });
+  if (objectLocation){
+    // Item found in a compendium -> get table data
+    return await game.packs.get(objectLocation).getDocument(id_uuid);
+  }else{
+    //it is a world item.
+    switch (type) {
+      case "RollTable":
+        return game.tables.filter(i=> i.id == id_uuid)[0];
+      case "Item":
```

Comment:
what about actors, macros or other types of compendium contents?
If this is just for tables maybe the function should be renamed.

Reply:
its for any type of element but since it was only used for tables and items thats what i coded, and sice we should start using the UUID this should not be a problem, but other types are easyle added.

Reply:
It looks much better :)